src/python/py_generate_pipeline.cpp

```diff
@@ -488,7 +491,12 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
      // Binding for GenerationConfig
     py::class_<GenerationConfig>(m, "GenerationConfig", generation_config_docstring)
         .def(py::init<std::string>(), py::arg("json_path"), "path where generation_config.json is stored")
-        .def(py::init([](py::kwargs kwargs) { return update_config_from_kwargs(GenerationConfig(), kwargs); }))
+        .def(py::init([](py::kwargs kwargs) {
+            if (kwargs.empty()) {
+                return GenerationConfig();
+            } else {
+                return *update_config_from_kwargs(GenerationConfig(), kwargs);
```

Comment:
is it possible to write `*update_config_from_kwargs({}, kwargs)` which mean optional does not have value?

to avoid if / else in current `init` method

Reply:
Yes, {} for empty value works. Removed redundant if.